### PR TITLE
Optimizations and supporting manual mappings

### PR DESCRIPTION
**Pull request by matteenm**
- This PR cleans up the code to be more extensible in supporting more chains in the future. 
- Supports using existing mappings in the original L1 list as the source of truth, using those values over any fetched ones for a given token. This allows manual entry/overrides of any mappings, even those on chains where fetching is not directly supported in this library (ex. Celo). 
- Uses different public optimism RPC endpoint (ankr). previous one seemed to get rate limited often, resulting in missing mappings. 
- Adds auto-formatting from prettier library. 

Tested by comparing output of this updated library with previous one and seeing no difference. Then adding some manual mappings to the input L1 list and seeing the correct additions to the output.

### CODE REVIEW

File: src/providers/index.ts [hunk 1]
```diff
@@ -112,6 +137,74 @@ function getMappingProvider(chainId: ChainId, l1TokenList: TokenList) {
...
+  chainIdToMappingsMap: {
+    [key: number]: PolygonMappedTokenData | GenericMappedTokenData
+  }
+): Promise<{
+  childTokenValid: boolean
+  childTokenAddress: string | undefined
+}> {
+  const existingMapping: undefined | string =
+    l1Token?.extensions?.bridgeInfo?.[chainId]?.tokenAddress
+  // use the externally fetched mappings if manual entry doesnt exist for the token/chain mapping
```
Comment (justindomingue) [inline src/providers/index.ts:177]:
```suggestion
  // use the externally fetched mappings if manual entry doesn't exist for the token/chain mapping
```

File: src/providers/index.ts [hunk 2]
```diff
@@ -112,6 +137,74 @@ function getMappingProvider(chainId: ChainId, l1TokenList: TokenList) {
...
+        (typeof childToken === 'object' ? !childToken.deleted : true) &&
+        (await hasExistingTokenContract(childTokenAddress, chainId))
+    )
+    return {
+      childTokenValid: childTokenValid,
+      childTokenAddress: childTokenAddress,
+    }
+  }
+  return {
+    childTokenValid: existingMapping ? true : false,
```
Comment (justindomingue) [inline src/providers/index.ts:199]:
```suggestion
    childTokenValid: !!existingMapping,
```

File: src/providers/index.ts [hunk 3]
```diff
@@ -112,6 +137,74 @@ function getMappingProvider(chainId: ChainId, l1TokenList: TokenList) {
...
+    childTokenAddress: existingMapping,
+  }
+}
+
+function validateChains(l2ChainIds: ChainId[]) {
```
Comment (justindomingue) [inline src/providers/index.ts:204]: is this called anywhere?

**Review by justindomingue — APPROVED** [review]
overall lgtm

File: src/providers/index.ts (hunk 3 above)
Comment (matteenm) [reply src/providers/index.ts:204]: ah no forgot to remove it. will do.

**Review by chunxi — APPROVED** [review]

**Pull request merged**